### ISSUE TITLE
Networking config per-cell

### ISSUE THREAD

**Issue by mattyg** [holochain/holochain#4350]
We will need to able to set some networking config per-cell. For example arc-clamping.

My goal is to publish backup data to multiple DHTs, but only store one copy of that data locally.

**Comment by neonphog** [comment]
@mattyg - Is this just about the arc-clamping, or are there additional configs? There is currently only one networking module per DNA, so some settings will currently not be able to be configured at the cell-level, but only at the DNA level. Arc clamping may be able to be per cell. So we need an explicit list to be able to judge which are possible.

**Comment by mattyg** [comment]
> @mattyg - Is this just about the arc-clamping, or are there additional configs? There is currently only one networking module per DNA, so some settings will currently not be able to be configured at the cell-level, but only at the DNA level. Arc clamping may be able to be per cell. So we need an explicit list to be able to judge which are possible.

Sorry yeah this is too vague. Let me think a bit more and tease it out. I should probably split it into multiple issues so we can get the simplest implementation in while laying groundwork for having cell-level config that can be updated at runtime.

**Comment by ThetaSinner** [comment]
My thoughts on this.

Most, if not all, instance configuration options can be kept so Kitsune can have defaults for settings that aren't overridden by an app. Bandwidth limits for gossip would be a good example of this, where most apps won't need to override the defaults.

Other configuration options should be selectively made available to apps. For example `gossip_redundancy_target`, because that is dependent on the expected network size and more generally by the use-case. 

For configuration options available to apps, I don't think it makes sense to have the tuning parameters be settable in the DNA manifest. That doesn't really guarantee that other nodes are respecting those settings because it's up to the code they happen to be running as to whether those parameters are really being enforced.
So these belong at the hApp manifest level I believe. 
That creates an interesting problem with the way that Kitsune works though. Because Kitsune operates at the DNA (space) granularity, if you have the same DNA shared between apps with different settings, which one do you respect?

**Comment by mattyg** [comment]
> Other configuration options should be selectively made available to apps. For example gossip_redundancy_target, because that is dependent on the expected network size and more generally by the use-case.
> For configuration options available to apps, I don't think it makes sense to have the tuning parameters be settable in the DNA manifest. That doesn't really guarantee that other nodes are respecting those settings because it's up to the code they happen to be running as to whether those parameters are really being enforced.
> So these belong at the hApp manifest level I believe.

Yes that makes sense -- as long as they are specified per `role` in the happ manifest.

>  That creates an interesting problem with the way that Kitsune works though. Because Kitsune operates at the DNA (space) granularity, if you have the same DNA shared between apps with different settings, which one do you respect?

I guess the options are:
- first-installed app wins
- kitsune defines a space by a dna+network settings so they get instantiated separately

**Comment by mattyg** [comment]
The only networking config that is clearly needed is for arc clamping. I extracted that to #4452.

Keeping this ticket open for discussion other cell / dna level networking config that might be needed.

@zippy I know we started talking about making bootstrap, signal, ice servers configured per-cell. Do you want to elaborate on the case for that here?

**Comment by mattyg** [comment]
Going to close this as there is no more discussion or pulls right now for additional per cell config.
